plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -164,8 +172,18 @@ public class OracleClient
 
     private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("uuuu-MM-dd");
     private static final DateTimeFormatter TIMESTAMP_SECONDS_FORMATTER = DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss");
-    private static final DateTimeFormatter TIMESTAMP_MILLIS_FORMATTER = DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss.SSS");
 
+    private static final Map<Integer, DateTimeFormatter> DATE_FORMATTER_MAP = Map.of(
+            0, DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss"),
+            1, DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss.S"),
+            2, DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss.SS"),
+            3, DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss.SSS"),
+            4, DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss.SSSS"),
+            5, DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss.SSSSS"),
+            6, DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss.SSSSSS"),
+            7, DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss.SSSSSSS"),
+            8, DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss.SSSSSSSS"),
+            9, DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss.SSSSSSSSS"));
```

Comment:
Is it possible to use `DateTimeFormatterBuilder` with `optionalStart()` , `appendFraction` & `optionalEnd`?

Reply:
refactored